From f6d407231423efd27330630542ea55c3241606dc Mon Sep 17 00:00:00 2001
From: fosero <fosero@users.noreply.github.com>
Date: Sun, 5 Jun 2016 16:23:50 +0200
Subject: [PATCH] ot-gpg-utils: use gentoo include path

---
 src/libotutil/ot-gpg-utils.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libotutil/ot-gpg-utils.h b/src/libotutil/ot-gpg-utils.h
index edb249f..55082c5 100644
--- a/src/libotutil/ot-gpg-utils.h
+++ b/src/libotutil/ot-gpg-utils.h
@@ -20,7 +20,7 @@
 
 #pragma once
 
-#include <gpgme.h>
+#include <gpgme/gpgme.h>
 #include <gio/gio.h>
 
 G_BEGIN_DECLS
-- 
2.7.3

